pkg/actions/lua_test.go

```diff
@@ -599,3 +453,429 @@ func TestDescendArgs(t *testing.T) {
 		}
 	})
 }
+
+// TestLuaRun_LakeFS tests the lakefs Lua module.
+func TestLuaRun_LakeFS(t *testing.T) {
+	// testing server for lakefs api, we use the request validator middleware to validate the requests
+	lakeFSServer := &testLakeFSServer{}
+	swagger, err := apigen.GetSwagger()
+	if err != nil {
+		t.Fatalf("failed to load swagger spec: %s", err)
+	}
+	router := chi.NewRouter()
+	router.Use(api.OapiRequestValidatorWithOptions(swagger, &openapi3filter.Options{
+		AuthenticationFunc: openapi3filter.NoopAuthenticationFunc,
+	}))
+
+	handler := apigen.HandlerFromMuxWithBaseURL(lakeFSServer, router, "/api/v1")
+	ts := httptest.NewServer(handler)
+	defer ts.Close()
+
+	tests := []struct {
+		Name            string
+		Script          string
+		ShouldFail      bool
+		ExpectedErr     bool
+		ExpectedOutput  string
+		ExpectedRequest map[string]any
+	}{
+		{
+			Name: "update_object_user_metadata",
+			Script: `local lakefs = require("lakefs")
+local code, resp = lakefs.update_object_user_metadata("repo", "branch", "path/to/object", {key="value", key2="value2"})
+print(code .. " " .. resp)
+`,
+			ExpectedOutput: "201",
+			ExpectedRequest: map[string]any{
+				"repository": "repo",
+				"branch":     "branch",
+				"params": apigen.UpdateObjectUserMetadataParams{
+					Path: "path/to/object",
+				},
+				"body": apigen.UpdateObjectUserMetadataJSONRequestBody{
+					Set: apigen.ObjectUserMetadata{
+						AdditionalProperties: map[string]string{
+							"key":  "value",
+							"key2": "value2",
+						},
+					},
+				},
+			},
+		},
+		{
+			Name: "update_object_user_metadata-no_meta",
+			Script: `local lakefs = require("lakefs")
+local code, resp = lakefs.update_object_user_metadata("repo", "branch", "object", nil)
+print(code .. " " .. resp)
+`,
+			ExpectedErr: true,
```

Comment:
Can we expect a more precise error?

Reply:
In the case of a Lua webhook, it is considered **either** successful **or** unsuccessful. 

Currently, we just log the error and consider the hook successful in all other cases.